include/irr/asset/IAsset.h

```diff
@@ -21,9 +21,51 @@ class IAssetMetadata : public core::IReferenceCounted
     virtual const char* getLoaderName() = 0;
 };
 
+//! An interface class representing any kind of cpu-objects with capability of being cached
+/**
+	Actually it can be anything like cpu-side meshes scenes, texture data and material pipelines, 
+	but must be serializable into/from .baw file format, unless it comes from an extension (irr::ext), 
+	so forth cached. There are different asset types you can find at IAsset::E_TYPE. 
+	IAsset doesn't provide direct instantiation (virtual destructor), much like IReferenceCounted.
+
+	Every asset must be loaded by a particular class derived from IAssetLoader.
+	
+	@see IAssetLoader
+	@see IAssetManager
+	@see IAssetWriter
+	@see IReferenceCounted
+*/
+
 class IAsset : virtual public core::IReferenceCounted
 {
 public:
+
+	/**
+		Values of E_TYPE represents an Asset type.
+
+		Entire enum as type represents a register with bit flags,
+		which represent an available Asset type.
+
+		@see IAsset
+
+		ET_BUFFER represents asset::ICPUBuffer
```

Comment:
if `IAsset* asset;` and `asset->getType()==ET_BUFFER` then you can `static_cast<ICPUBuffer*>(asset)` safely

Reply:
nah, never use reinterpret_cast unless it's an absolute must

Reply:
`static_cast` works beautifully because it can calculate the relative offset to the pointer accounting for the difference between the start of storage of a base class and a derived class.

Example
```cpp
class A {uint32_t a;};
class B {uint32_t b;};

class C : public A, public B {};

B* base;
C* derived = static_cast<C*>(base);
```


if you print both `base` and `derived` as `%p` you will see that the pointer to derived has been rolled back by 4 bytes.


`reinterpret_cast` is only for casting void pointers really.

Reply:
status?

Reply:
Solved